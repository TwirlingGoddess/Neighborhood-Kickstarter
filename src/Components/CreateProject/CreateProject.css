.create-project-section {
  align-items: center;
  display: flex;
  height: 100vh;
  flex-direction: column;
  width: 100%;
  justify-content: center;
}

.main-section {
  display: flex;
  justify-content: center;
  align-items: center;
}

.listed-materials {
  display: flex;
  width: 30rem;
  margin: 2rem;
  height: 35rem;
  background-image: linear-gradient(to top, #09203f 0%, #537895 100%);
  flex-direction: column;
  border-radius: 0.7em;
  overflow-wrap: break-word;
  overflow: scroll;
}

.list-item {
  max-width: 30rem;
  font-size: 1.9rem;
  margin: .4rem;
}

.create-project-form {
  display: flex;
  align-items: center;
  border-radius: 0.7em;
  background-image: linear-gradient(to top, #09203f 0%, #537895 100%);
  justify-content: flex-start;
  width: 30rem;
  height: 35rem;
  flex-direction: column;
}

.project-inputs {
  margin: .3rem;
  color: black;
  padding-left: .4rem;
  width: 20rem;
  height: 2.5rem;
  font-size: 1.5rem;
  background-image: linear-gradient(-225deg, #FFFEFF 0%, #D7FFFE 100%);
  border-radius: 0.6em;
  border: none;
}

.materials {
  height: 5rem;
}

.description {
  height: 5rem;
}

.add-material-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin-top: 25px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 19.8rem;
  max-height: 4rem;
}

.add-material-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}

.submit-project-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin-top: 25px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 19.8rem;
  max-height: 4rem;
}

.submit-project-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}

.view-projects-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin: 20px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 12rem;
  max-height: 4rem;
}

.view-projects-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}

.delete-material {
  margin-left: .4rem;
  width: 1rem;
  height: 1rem;
}

.delete-material {
  cursor: pointer;
}

.create-project-form label {
  color: white;
}
.create-project-form input {
  color: white;
}
