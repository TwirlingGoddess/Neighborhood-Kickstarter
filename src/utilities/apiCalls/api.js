var fs = require('fs');
var S3FS = require('s3fs');
var s3fsImpl = new S3FS('leeleestestbucket1', {
  accessKeyId: 'AKIAIXU4Q4Y4XZWUK4SQ', 
  secretAccessKey: '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'
});

s3fsImpl.create();

var multiparty = require('connect-multiparty'),
  multipartyMiddleware = multiparty();

module.exports = function(router, passport){
  router.use(multipartyMiddleware);

  router.use(passport.authenticate('bearer', { session: false}));
  router.use(function(req, res, next){
    fs.appendFile('log.txt' req.path + " token: " + req.query.access_token + " ")
      function(err){
        next();
      });
  });

  router.post('/testupload', function(req, res){
    var file = req.files.file;
    var stream = fs.createReadStream(file.path);
    return s3fsImpl.writeFile(file.originalFilename, stream).then(function(){
      fs.unlink(file.path, function(err){
        if(err){
          console.error(err)
        }
      })
      res.redirect('/profile');
    })
  })

  router.get('/testAPI', function(req, res){
    res.json({ SecretData: 'abc123' });
  });


}