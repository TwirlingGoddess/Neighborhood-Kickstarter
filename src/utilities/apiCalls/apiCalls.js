export const getNeighborhoods = async () => {
  const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods');
  const neighborhoods = await response.json();
  return neighborhoods
}

export const getProjects = async () => {
  const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/projects');
  const projects = await response.json();
  return projects
}

export const getNeighborhoodProjectsById = async (id) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods/${id}`);
  const neighborhoodProjects = await response.json();
  return neighborhoodProjects
}

export const getUserProjects = async (userId) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${userId}/projects`);
  const userProjects = await response.json();
  return userProjects
}

export const getAllUsers = async () => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users`);
  const users = await response.json();
  return users
}

export const getProjectComments = async (id) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/projects/${id}/comments`)
  const comments = response.json();
  return comments
}

export const addNewUser = async (user) => {
  console.log(JSON.stringify(user))
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/google_users`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(user)
    })
    const currentUser = await response.json();
    return currentUser
  } catch (error) {
    throw new Error(error.message)
  }
}

export const addNewUserLocal = async (user) => {
  console.log(JSON.stringify(user))
  try {
    const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/users', {
      method: 'POST',
      headers: {
        'Content-type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify(user)
    })
    const currentUser = await response.json();
    return currentUser
  } catch (error) {
    throw new Error(error.message)
  }
}

export const postNewProject = async (newProject, id) => {
  try {
    await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${id}/projects`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newProject)
    })
  } catch (error) {
    throw new Error(error.message);
  }
}