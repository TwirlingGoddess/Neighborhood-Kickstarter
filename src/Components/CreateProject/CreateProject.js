import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './CreateProject.css';
import deleteButton from '../../images/x-button.svg';
import { postNewProject } from '../../utilities/apiCalls/apiCalls';

class CreateProject extends Component {
  constructor() {
    super();
    this.state = {
      currentUserId:'',
      title: '',
      description: '',
      newResource: '',
      resources: [],
      photo: null,
    }
  }

  componentDidMount = () => {
    this.updateCurrentUser();
  }

  updateCurrentUser = () => {
    let currentUserId = this.props.currentUser.id;
    console.log(currentUserId)

    if(currentUserId) {
      this.setState({
        currentUserId
      })
    }
    return 
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    })
  }

  handleSubmit = async (event) => {
    event.preventDefault();

    if(event.target.className === 'submit-project-button') {
      const { title, description, resources, currentUserId, photo} = this.state; 

      const newProject = {
        project: {
          title,
          description,
          photo,
          resources
        }
      };
      console.log(newProject)

      await postNewProject(newProject, currentUserId);

      this.setState({
        title: '',
        description: '',
        newResource: '',
        resources: [],
        photo: null,
      })
    }
    return
  }

  addResource = (event, resource) => {
    event.preventDefault();
    if(event.target.className === 'add-material-button') {
      let newResource = {name: resource};
      let resources = [...this.state.resources, newResource];
  
      this.setState({
        resources, 
        newResource: ''
      })
    }
  }

  handlePhoto = async (event) => {
    const image = event.target.files[0]

    const xmlImage = `
    <Contents>
        <Key>Screen Shot 2018-05-01 at 12.24.10 PM copy.png</Key>
        <LastModified>2018-10-31T21:48:52.000Z</LastModified>
        <ETag>&quot;45edc57f143cf68cc42c3b1a0341958f&quot;</ETag>
        <Size>521651</Size>
        <StorageClass>STANDARD</StorageClass>
    </Contents>
    `
    const url = 'http://neighbor-hub-images.s3.amazonaws.com'
    const settings = {
      method: 'POST'
      ,
      headers: {
        Accept: 'application/json',
        // 'Content-Type': 'multipart/form-data',
        'Access-Control-Allow-Origin': '*'
        },
      body: xmlImage
    }

    try{
      const response = await fetch(url, settings);
      console.log(response)
      const data = await response.text();
      console.log(data)
      const string = await (new window.DOMParser()).parseFromString(data, "text/xml")
      console.log(string)

        if (response.ok) {
          console.log(response)
            // return response
        } else {
            console.log('response is not ok')
        }
    } catch (error) {
        console.log(error.message)
    }

  }

  deleteMaterial = (item) => {
    let resources = this.state.resources.filter(resource => {
      return resource.name !== item;
    })
    this.setState({
      resources
    })
  }

  render() {
    return(
      <div className='create-project-section'>
        <NavLink className='view-projects-button' to='/Landing'>View Projects</NavLink>
        <div className='main-section'>
          <form className='create-project-form'>
            <h1>Create Project</h1>
            <input
              className='project-inputs'
              name='title'
              type='text'
              placeholder='Project Name'
              value={this.state.title}
              onChange={this.handleChange}
            />
            <textarea
              className='project-inputs description'
              name='description'
              type='text'
              placeholder='Project description:'
              value={this.state.description}
              onChange={this.handleChange}
            />
            <textarea
              className='project-inputs materials'
              name='newResource'
              type='text'
              placeholder='Add new material'
              value={this.state.newResource}
              onChange={this.handleChange}
            />          
            <label htmlFor="choose-file" className="file-label">Upload Image</label>
            <input 
              className="choose-file" 
              type="file"
              value=''
              onChange={this.handlePhoto}
            />
            <button className='add-material-button' onClick={(event) => this.addResource(event, this.state.newResource)}>Add Resource</button>
            <button className='submit-project-button' onClick={this.handleSubmit}>Submit Project</button>
          </form>

          <div className='listed-materials'>
            <h1>Materials Needed:</h1>
              {this.state.resources.map((resource, index) => {
                return <div 
                        key={index}
                        className='list-item'>{`${index+1}. ${resource.name}`}
                        <img onClick={() => this.deleteMaterial(resource.name)} alt='remove button' className='delete-material' src={deleteButton}/>
                      </div>
              })}
          </div>
        </div>
      </div>
    )
  }
}

export default CreateProject;