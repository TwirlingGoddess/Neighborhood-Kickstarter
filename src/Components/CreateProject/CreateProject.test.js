import React from 'react';
import CreateProject from './CreateProject';
import { postNewProject } from '../../utilities/apiCalls/apiCalls';
import { shallow } from 'enzyme';

describe('CreateProject', () => {
  let wrapper;
  let mockCurrentUser = {};

  it('should match the snapshot', () => {
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with the current user id if there is a current user', () => {
    mockCurrentUser = {id: 1};
    let id = mockCurrentUser.id;
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.instance().updateCurrentUser();
    expect(wrapper.state('currentUserId')).toEqual(id);
  });

  it('should not update state if there is not a current user', () => {
    mockCurrentUser = {};
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.instance().updateCurrentUser();
    expect(wrapper.state('currentUserId')).toEqual('');
  });

  it('should update state with the input information', () => {
    mockCurrentUser = {id: 1};
    let mockEvent = {target: {name: 'title', value: 'Yaassss'}};
    let userInput = 'Yaassss';
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('title')).toEqual(userInput);
  });

  it('should call handleSubmit when the submit button is clicked', () => {
    const mockUser = {id: 1}
    let mockedEvent = { target: {className: 'add-material-button'}, preventDefault: () => {} };
    let newResource = 'wood';
    let resource = { name: newResource};
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.setState({ newResource })
    wrapper.find('.submit-project-button').simulate('submit')
    wrapper.instance().handleSubmit(mockedEvent, resource)
    expect(wrapper.state('resources')).toHaveLength(0);
  })

  it('should return data from postNewProject fetch', async () => {
    const expected = undefined
    mockCurrentUser = {id: 1};
    let newResource = 'wood';
    let resource = { name: newResource};
    const mockProject= {
          title: 'Love',
          description: 'Never lasts',
          photo: 'https://apod.nasa.gov/apod/image/1811/CaliforniaNebula_Falls_960.jpg',
          resources: [ resource, resource ]
        }    
    let data = await postNewProject(mockProject, mockCurrentUser);
    expect(data).toEqual(undefined);
  });

  it('should add a resource if the submit resource button is clicked', () => {
    mockCurrentUser = {id: 1};
    let mockedEvent = { target: {className: 'add-material-button'}, preventDefault: () => {} };
    let newResource = 'wood';
    let resource = { name: newResource};
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.setState({ newResource });
    wrapper.instance().addResource(mockedEvent, resource);
    expect(wrapper.state('resources')).toHaveLength(1);
  });

  it('should delete a resource if the delete button is clicked', () => {
    mockCurrentUser = {id: 1};
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    wrapper.setState({resources: [{name: 'wood'}]});
    expect(wrapper.state('resources')).toHaveLength(1);
    wrapper.find('.delete-material').simulate('click');
    expect(wrapper.state('resources')).toHaveLength(0);
  });

  it('should upload an image file when file upload input is clicked', () => {
    mockCurrentUser = {id: 1};
    const initial = ''
    const expected = 'https://apod.nasa.gov/apod/image/1811/CaliforniaNebula_Falls_960.jpg'
    wrapper = shallow(<CreateProject currentUser={mockCurrentUser}/>);
    expect(wrapper.state('photo')).toEqual(initial);
    wrapper.find('.choose-file').simulate('click');
    wrapper.setState({photo: expected});
    expect(wrapper.state('photo')).toEqual(expected);
  })

}); 