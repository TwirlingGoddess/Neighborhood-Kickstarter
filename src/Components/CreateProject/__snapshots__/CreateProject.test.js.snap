// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProject should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateProject
    currentUser={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="view-projects-button"
          to="/Landing"
        >
          View Projects
        </NavLink>,
        <div
          className="main-section"
        >
          <form
            className="create-project-form"
          >
            <h1>
              Create Project
            </h1>
            <input
              className="project-inputs"
              name="title"
              onChange={[Function]}
              placeholder="Project Name"
              type="text"
              value=""
            />
            <textarea
              className="project-inputs description"
              name="description"
              onChange={[Function]}
              placeholder="Project description:"
              type="text"
              value=""
            />
            <textarea
              className="project-inputs materials"
              name="newResource"
              onChange={[Function]}
              placeholder="Add new material"
              type="text"
              value=""
            />
            <label
              className="file-label"
              htmlFor="choose-file"
            >
              Upload Image
            </label>
            <input
              action="http://neighbor-hub-images.s3.amazonaws.com/"
              className="choose-file"
              encType="multipart/form-data"
              method="post"
              type="file"
              value=""
            />
            <button
              className="add-material-button"
              onClick={[Function]}
            >
              Add Resource
            </button>
            <label>
              Pick An Image
            </label>
            <input
              accept="image/*"
              onChange={[Function]}
              type="file"
            />
            <button
              className="submit-project-button"
              onClick={[Function]}
            >
              Submit Project
            </button>
          </form>
          <div
            className="listed-materials"
          >
            <h1>
              Materials Needed:
            </h1>
          </div>
        </div>,
      ],
      "className": "create-project-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "View Projects",
          "className": "view-projects-button",
          "to": "/Landing",
        },
        "ref": null,
        "rendered": "View Projects",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="create-project-form"
            >
              <h1>
                Create Project
              </h1>
              <input
                className="project-inputs"
                name="title"
                onChange={[Function]}
                placeholder="Project Name"
                type="text"
                value=""
              />
              <textarea
                className="project-inputs description"
                name="description"
                onChange={[Function]}
                placeholder="Project description:"
                type="text"
                value=""
              />
              <textarea
                className="project-inputs materials"
                name="newResource"
                onChange={[Function]}
                placeholder="Add new material"
                type="text"
                value=""
              />
              <label
                className="file-label"
                htmlFor="choose-file"
              >
                Upload Image
              </label>
              <input
                action="http://neighbor-hub-images.s3.amazonaws.com/"
                className="choose-file"
                encType="multipart/form-data"
                method="post"
                type="file"
                value=""
              />
              <button
                className="add-material-button"
                onClick={[Function]}
              >
                Add Resource
              </button>
              <label>
                Pick An Image
              </label>
              <input
                accept="image/*"
                onChange={[Function]}
                type="file"
              />
              <button
                className="submit-project-button"
                onClick={[Function]}
              >
                Submit Project
              </button>
            </form>,
            <div
              className="listed-materials"
            >
              <h1>
                Materials Needed:
              </h1>
            </div>,
          ],
          "className": "main-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Create Project
                </h1>,
                <input
                  className="project-inputs"
                  name="title"
                  onChange={[Function]}
                  placeholder="Project Name"
                  type="text"
                  value=""
                />,
                <textarea
                  className="project-inputs description"
                  name="description"
                  onChange={[Function]}
                  placeholder="Project description:"
                  type="text"
                  value=""
                />,
                <textarea
                  className="project-inputs materials"
                  name="newResource"
                  onChange={[Function]}
                  placeholder="Add new material"
                  type="text"
                  value=""
                />,
                <label
                  className="file-label"
                  htmlFor="choose-file"
                >
                  Upload Image
                </label>,
                <input
                  action="http://neighbor-hub-images.s3.amazonaws.com/"
                  className="choose-file"
                  encType="multipart/form-data"
                  method="post"
                  type="file"
                  value=""
                />,
                <button
                  className="add-material-button"
                  onClick={[Function]}
                >
                  Add Resource
                </button>,
                <label>
                  Pick An Image
                </label>,
                <input
                  accept="image/*"
                  onChange={[Function]}
                  type="file"
                />,
                <button
                  className="submit-project-button"
                  onClick={[Function]}
                >
                  Submit Project
                </button>,
              ],
              "className": "create-project-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create Project",
                },
                "ref": null,
                "rendered": "Create Project",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-inputs",
                  "name": "title",
                  "onChange": [Function],
                  "placeholder": "Project Name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-inputs description",
                  "name": "description",
                  "onChange": [Function],
                  "placeholder": "Project description:",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "project-inputs materials",
                  "name": "newResource",
                  "onChange": [Function],
                  "placeholder": "Add new material",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Upload Image",
                  "className": "file-label",
                  "htmlFor": "choose-file",
                },
                "ref": null,
                "rendered": "Upload Image",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "action": "http://neighbor-hub-images.s3.amazonaws.com/",
                  "className": "choose-file",
                  "encType": "multipart/form-data",
                  "method": "post",
                  "onChange": undefined,
                  "type": "file",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Resource",
                  "className": "add-material-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add Resource",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pick An Image",
                },
                "ref": null,
                "rendered": "Pick An Image",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "accept": "image/*",
                  "onChange": [Function],
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit Project",
                  "className": "submit-project-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Submit Project",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Materials Needed:
                </h1>,
                Array [],
              ],
              "className": "listed-materials",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Materials Needed:",
                },
                "ref": null,
                "rendered": "Materials Needed:",
                "type": "h1",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="view-projects-button"
            to="/Landing"
          >
            View Projects
          </NavLink>,
          <div
            className="main-section"
          >
            <form
              className="create-project-form"
            >
              <h1>
                Create Project
              </h1>
              <input
                className="project-inputs"
                name="title"
                onChange={[Function]}
                placeholder="Project Name"
                type="text"
                value=""
              />
              <textarea
                className="project-inputs description"
                name="description"
                onChange={[Function]}
                placeholder="Project description:"
                type="text"
                value=""
              />
              <textarea
                className="project-inputs materials"
                name="newResource"
                onChange={[Function]}
                placeholder="Add new material"
                type="text"
                value=""
              />
              <label
                className="file-label"
                htmlFor="choose-file"
              >
                Upload Image
              </label>
              <input
                action="http://neighbor-hub-images.s3.amazonaws.com/"
                className="choose-file"
                encType="multipart/form-data"
                method="post"
                type="file"
                value=""
              />
              <button
                className="add-material-button"
                onClick={[Function]}
              >
                Add Resource
              </button>
              <label>
                Pick An Image
              </label>
              <input
                accept="image/*"
                onChange={[Function]}
                type="file"
              />
              <button
                className="submit-project-button"
                onClick={[Function]}
              >
                Submit Project
              </button>
            </form>
            <div
              className="listed-materials"
            >
              <h1>
                Materials Needed:
              </h1>
            </div>
          </div>,
        ],
        "className": "create-project-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "View Projects",
            "className": "view-projects-button",
            "to": "/Landing",
          },
          "ref": null,
          "rendered": "View Projects",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                className="create-project-form"
              >
                <h1>
                  Create Project
                </h1>
                <input
                  className="project-inputs"
                  name="title"
                  onChange={[Function]}
                  placeholder="Project Name"
                  type="text"
                  value=""
                />
                <textarea
                  className="project-inputs description"
                  name="description"
                  onChange={[Function]}
                  placeholder="Project description:"
                  type="text"
                  value=""
                />
                <textarea
                  className="project-inputs materials"
                  name="newResource"
                  onChange={[Function]}
                  placeholder="Add new material"
                  type="text"
                  value=""
                />
                <label
                  className="file-label"
                  htmlFor="choose-file"
                >
                  Upload Image
                </label>
                <input
                  action="http://neighbor-hub-images.s3.amazonaws.com/"
                  className="choose-file"
                  encType="multipart/form-data"
                  method="post"
                  type="file"
                  value=""
                />
                <button
                  className="add-material-button"
                  onClick={[Function]}
                >
                  Add Resource
                </button>
                <label>
                  Pick An Image
                </label>
                <input
                  accept="image/*"
                  onChange={[Function]}
                  type="file"
                />
                <button
                  className="submit-project-button"
                  onClick={[Function]}
                >
                  Submit Project
                </button>
              </form>,
              <div
                className="listed-materials"
              >
                <h1>
                  Materials Needed:
                </h1>
              </div>,
            ],
            "className": "main-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Create Project
                  </h1>,
                  <input
                    className="project-inputs"
                    name="title"
                    onChange={[Function]}
                    placeholder="Project Name"
                    type="text"
                    value=""
                  />,
                  <textarea
                    className="project-inputs description"
                    name="description"
                    onChange={[Function]}
                    placeholder="Project description:"
                    type="text"
                    value=""
                  />,
                  <textarea
                    className="project-inputs materials"
                    name="newResource"
                    onChange={[Function]}
                    placeholder="Add new material"
                    type="text"
                    value=""
                  />,
                  <label
                    className="file-label"
                    htmlFor="choose-file"
                  >
                    Upload Image
                  </label>,
                  <input
                    action="http://neighbor-hub-images.s3.amazonaws.com/"
                    className="choose-file"
                    encType="multipart/form-data"
                    method="post"
                    type="file"
                    value=""
                  />,
                  <button
                    className="add-material-button"
                    onClick={[Function]}
                  >
                    Add Resource
                  </button>,
                  <label>
                    Pick An Image
                  </label>,
                  <input
                    accept="image/*"
                    onChange={[Function]}
                    type="file"
                  />,
                  <button
                    className="submit-project-button"
                    onClick={[Function]}
                  >
                    Submit Project
                  </button>,
                ],
                "className": "create-project-form",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create Project",
                  },
                  "ref": null,
                  "rendered": "Create Project",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-inputs",
                    "name": "title",
                    "onChange": [Function],
                    "placeholder": "Project Name",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-inputs description",
                    "name": "description",
                    "onChange": [Function],
                    "placeholder": "Project description:",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "project-inputs materials",
                    "name": "newResource",
                    "onChange": [Function],
                    "placeholder": "Add new material",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Upload Image",
                    "className": "file-label",
                    "htmlFor": "choose-file",
                  },
                  "ref": null,
                  "rendered": "Upload Image",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "action": "http://neighbor-hub-images.s3.amazonaws.com/",
                    "className": "choose-file",
                    "encType": "multipart/form-data",
                    "method": "post",
                    "onChange": undefined,
                    "type": "file",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Resource",
                    "className": "add-material-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Add Resource",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pick An Image",
                  },
                  "ref": null,
                  "rendered": "Pick An Image",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "accept": "image/*",
                    "onChange": [Function],
                    "type": "file",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit Project",
                    "className": "submit-project-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Submit Project",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Materials Needed:
                  </h1>,
                  Array [],
                ],
                "className": "listed-materials",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Materials Needed:",
                  },
                  "ref": null,
                  "rendered": "Materials Needed:",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
